helpers.go

```diff
@@ -331,21 +331,27 @@ func (iter *Iter) RowData() (RowData, error) {
 	values := make([]interface{}, 0, len(iter.Columns()))
 
 	for _, column := range iter.Columns() {
-		if c, ok := column.TypeInfo.(TupleTypeInfo); !ok {
-			val, err := column.TypeInfo.NewWithError()
-			if err != nil {
-				return RowData{}, err
-			}
+		if column.Name == "rack" && column.Keyspace == "system" && (column.Table == "peers_v2" || column.Table == "peers") {
```

Comment:
This seems like a hack, plus it could break backwards compatibility for existing users of `RowData` (outside of gocql) that read from those tables.
Could the issue be fixed another way, for example improving the API/adding new API for all users?

Reply:
There is #1834, which allows to handle a nullable value. After #1834 is merged, I will refactor this PR using the nullable scan functional. This should help to avoid the hack.

Reply:
I agree with addressing #1834 first, @martin-sucha can you provide feedback on #1834 ? I see [your original comment](https://github.com/apache/cassandra-gocql-driver/issues/1699#issuecomment-1556901238) on that issue, but I'm not sure if #1834 accurately implements what you were proposing.